### ISSUE TITLE
dw2pdf cannot export ⇒ symbol

### ISSUE THREAD

**Issue by radiks** [splitbrain/dokuwiki-plugin-dw2pdf#76]
Hi, 

Thanks for the plugin. It is very useful for me.

I noticed the following problem:
When I insert into a document, "<=" dokuwiki turning it into a ⇐ symbol. And when I export to PDF then there is a problem with the interpretation of this symbol.

Regards

**Comment by Klap-in** [comment]
Strange issue. It works here fine for me: symbols like ⇒ ⇐ and image replacements like FIXME.
So the interesting question is why is mis-behavior triggered?

Can you manually add `&debughtml=html` or maybe more handy `&debughtml=text` to the url of the pdf-export. And share me the html code around these symbols?
Eventually you may share an affected example pdf as well, however I have so far no tools to look into the source of a pdf...

**Comment by Juergen-aus-Koeln** [comment]
Sorry, I don't know how to extract and share the html code by the debug-commands. 

Perhaps this may help you: Instead of the debugging results I have uploaded this pdf-file:
![ideenpool-3 pdf](https://cloud.githubusercontent.com/assets/1800472/5682644/c6f68eaa-9821-11e4-95c2-579307f9ffa2.png)
Please copy it to the local file system, rename it from png to pdf and you can see the result.

Regards Juergen

**Comment by Klap-in** [comment]
to export the pdf you click probably on a button with a link ending with `do=export_pdf`.
e.g. `http://yourwiki.org/yourpage?do=export_pdf`
Adding debughtml=text e.g. `http://yourwiki.org/yourpage?do=export_pdf&debughtml=text`
Let you show the html which will converted to a pdf.

You can copy the relevant snippet and paste it here inside three ``` like:
 ```
  ```html
 (start with three back quotes)
  ...your snippet...
 (end with three back quotes, without spaces in between)
  `` `
```

**Comment by Klap-in** [comment]
Can you please include symbols like  ⇒ ⇐  as well?

**Comment by Juergen-aus-Koeln** [comment]
Here is the wiki code:
```
====== PlayGround ======

FIXME ⇒ :!: ⇐
```
The FIXME and the :!: are not supported in the PDF.

And here is the code of the whole little page (I don't know where it makes sense to separate the snippet):
```
<html><head><style type="text/css">div.error,
div.info,
div.success,
div.notify {
  color: #000;
  background-repeat: no-repeat;
  background-position: 8px 50%;
  border: 1px solid;
  font-size: 90%;
  margin: 0 0 0.5em;
  padding: 0.4em;
  padding-left: 32px;
  overflow: hidden;
  border-radius: 5px;
}
[dir=rtl] div.error,
[dir=rtl] div.info,
[dir=rtl] div.success,
[dir=rtl] div.notify {
  background-position: 99% 50%;
  padding-left: .4em;
  padding-right: 32px;
}
div.error {
  background-color: #fcc;
  background-image: url(/wiki/lib/styles/../images/error.png);
  border-color: #ebb;
}
div.info {
  background-color: #ccf;
  background-image: url(/wiki/lib/styles/../images/info.png);
  border-color: #bbe;
}
div.success {
  background-color: #cfc;
  background-image: url(/wiki/lib/styles/../images/success.png);
  border-color: #beb;
}
div.notify {
  background-color: #ffc;
  background-image: url(/wiki/lib/styles/../images/notify.png);
  border-color: #eeb;
}
.JSpopup,
#link__wiz {
  position: absolute;
  background-color: #fff;
  color: #000;
  z-index: 20;
  overflow: hidden;
}
#link__wiz .ui-dialog-content {
  padding-left: 0;
  padding-right: 0;
}
#media__popup_content button.button {
  border: 1px outset;
}
#media__popup_content button.selected {
  border-style: inset;
}
.a11y {
  position: absolute !important;
  left: -99999em !important;
  top: auto !important;
  width: 1px !important;
  height: 1px !important;
  overflow: hidden !important;
}
[dir=rtl] .a11y {
  left: auto !important;
  right: -99999em !important;
}
.code .co0 {
  color: #666666;
  font-style: italic;
}
.code .co4 {
  color: #cc0000;
  font-style: italic;
}
.code .es5 {
  color: #006699;
  font-weight: bold;
}
.code .es6 {
  color: #009933;
  font-weight: bold;
}
.code .kw2 {
  color: #000000;
  font-weight: bold;
}
.code .kw5 {
  color: #008000;
}
.code .kw6 {
  color: #f08;
  font-weight: bold;
}
.code .me0 {
  color: #004000;
}
.code .nu0 {
  color: #cc66cc;
}
.code .re0 {
  color: #0000ff;
}
.code .re3 {
  color: #ff3333;
  font-weight: bold;
}
.code .re4 {
  color: #009999;
}
.code .re5 {
  color: #660033;
}
.code .sc-2 {
  color: #404040;
}
.code .sy3 {
  color: #000040;
}
.code .br0,
.code .sy0 {
  color: #66cc66;
}
.code .co1,
.code .coMULTI,
.code .sc-1 {
  color: #808080;
  font-style: italic;
}
.code .co2,
.code .sy1 {
  color: #339933;
}
.code .co3,
.code .sy4 {
  color: #008080;
}
.code .es0,
.code .es1,
.code .esHARD {
  color: #000099;
  font-weight: bold;
}
.code .es2,
.code .es3,
.code .es4 {
  color: #660099;
  font-weight: bold;
}
.code .kw1,
.code .kw8 {
  color: #b1b100;
}
.code .kw10,
.code .kw11,
.code .kw12,
.code .kw9 {
  color: #003399;
  font-weight: bold;
}
.code .kw13,
.code .kw14,
.code .kw15,
.code .kw16,
.code .me1,
.code .me2 {
  color: #006600;
}
.code .kw3,
.code .kw7,
.code .sy2 {
  color: #000066;
}
.code .kw4,
.code .re2 {
  color: #993333;
}
.code .re1,
.code .st0,
.code .st_h {
  color: #ff0000;
}
div.error,
div.info,
div.success,
div.notify,
.secedit,
.a11y,
.JSpopup,
#link__wiz {
  display: none;
}
.dokuwiki .plugin_wrap {
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}
.dokuwiki .plugin_wrap table {
  width: 100%;
}
.dokuwiki .plugin_wrap em strong {
  font-size: 130%;
  font-weight: bold;
  font-style: normal;
  display: block;
}
.dokuwiki .plugin_wrap em strong em.u {
  font-size: 115%;
  border-bottom: 1px solid #D0D0D0;
  font-style: normal;
  text-decoration: none;
  display: block;
}
.dokuwiki .wrap_danger em strong em.u,
.dokuwiki .wrap_warning em strong em.u,
.dokuwiki .wrap_caution em strong em.u,
.dokuwiki .wrap_notice em strong em.u,
.dokuwiki .wrap_safety em strong em.u {
  text-transform: uppercase;
  border-bottom-width: 0;
}
.dokuwiki .wrap_box em strong em.u,
.dokuwiki .wrap_info em strong em.u,
.dokuwiki .wrap_important em strong em.u,
.dokuwiki .wrap_alert em strong em.u,
.dokuwiki .wrap_tip em strong em.u,
.dokuwiki .wrap_help em strong em.u,
.dokuwiki .wrap_todo em strong em.u,
.dokuwiki .wrap_download em strong em.u {
  border-bottom-color: #999;
}
.dokuwiki .plugin_wrap h1,
.dokuwiki .plugin_wrap h2,
.dokuwiki .plugin_wrap h3,
.dokuwiki .plugin_wrap h4,
.dokuwiki .plugin_wrap h5 {
  margin-left: 0;
  margin-right: 0;
}
.dokuwiki .wrap_left,
.dokuwiki .wrap_column {
  float: left;
  margin-right: 1.5em;
}
[dir=rtl] .dokuwiki .wrap_column {
  float: right;
  margin-left: 1.5em;
  margin-right: 0;
}
.dokuwiki .wrap_right {
  float: right;
  margin-left: 1.5em;
}
.dokuwiki .wrap_center {
  display: block;
  margin-left: auto;
  margin-right: auto;
}
.dokuwiki .wrap_col2,
.dokuwiki .wrap_col3,
.dokuwiki .wrap_col4,
.dokuwiki .wrap_col5 {
  -moz-column-gap: 1.5em;
  -webkit-column-gap: 1.5em;
  column-gap: 1.5em;
  -moz-column-rule: 1px dotted #666;
  -webkit-column-rule: 1px dotted #666;
  column-rule: 1px dotted #666;
}
.dokuwiki .wrap_col2 {
  -moz-column-count: 2;
  -webkit-column-count: 2;
  column-count: 2;
}
.dokuwiki .wrap_col3 {
  -moz-column-count: 3;
  -webkit-column-count: 3;
  column-count: 3;
}
.dokuwiki .wrap_col4 {
  -moz-column-count: 4;
  -webkit-column-count: 4;
  column-count: 4;
}
.dokuwiki .wrap_col5 {
  -moz-column-count: 5;
  -webkit-column-count: 5;
  column-count: 5;
}
.dokuwiki .wrap_half {
  width: 48%;
  margin-right: 4%;
}
.dokuwiki .wrap_third {
  width: 30%;
  margin-right: 5%;
}
.dokuwiki .wrap_quarter {
  width: 22%;
  margin-right: 4%;
}
[dir=rtl] .dokuwiki .wrap_half,
[dir=rtl] .dokuwiki .wrap_third,
[dir=rtl] .dokuwiki .wrap_quarter {
  margin-right: 0;
  margin-left: 4%;
}
[dir=rtl] .dokuwiki .wrap_third {
  margin-left: 5%;
}
.dokuwiki .wrap_half:nth-of-type(2n),
.dokuwiki .wrap_third:nth-of-type(3n),
.dokuwiki .wrap_quarter:nth-of-type(4n) {
  margin-right: 0;
}
[dir=rtl] .dokuwiki .wrap_half:nth-of-type(2n),
[dir=rtl] .dokuwiki .wrap_third:nth-of-type(3n),
[dir=rtl] .dokuwiki .wrap_quarter:nth-of-type(4n) {
  margin-left: 0;
}
.dokuwiki .wrap_half:nth-of-type(2n+1),
.dokuwiki .wrap_third:nth-of-type(3n+1),
.dokuwiki .wrap_quarter:nth-of-type(4n+1) {
  clear: left;
}
[dir=rtl] .dokuwiki .wrap_half:nth-of-type(2n+1),
[dir=rtl] .dokuwiki .wrap_third:nth-of-type(3n+1),
[dir=rtl] .dokuwiki .wrap_quarter:nth-of-type(4n+1) {
  clear: right;
}
@media only screen and (max-width: 600px) {
  .dokuwiki .wrap_half,
  .dokuwiki .wrap_third,
  .dokuwiki .wrap_quarter {
    width: auto;
    margin-right: 0;
    margin-left: 0;
    float: none;
  }
}
.dokuwiki .wrap_leftalign {
  text-align: left;
}
.dokuwiki .wrap_centeralign {
  text-align: center;
}
.dokuwiki .wrap_rightalign {
  text-align: right;
}
.dokuwiki .wrap_justify {
  text-align: justify;
}
.dokuwiki div.wrap_round {
  border-radius: 1.4em;
}
.dokuwiki span.wrap_round {
  border-radius: .14em;
}
.dokuwiki .wrap_lo {
  color: #666;
  font-size: 85%;
}
.dokuwiki .wrap_em {
  color: #c00;
  font-weight: bold;
}
.dokuwiki .wrap_indent {
  padding-left: 1.5em;
}
[dir=rtl] .dokuwiki .wrap_indent {
  padding-right: 1.5em;
  padding-left: 0;
}
.dokuwiki .wrap_outdent {
  margin-left: -1.5em;
}
[dir=rtl] .dokuwiki .wrap_outdent {
  margin-right: -1.5em;
  margin-left: 0;
}
.dokuwiki div.wrap_prewrap pre {
  white-space: pre-wrap;
  word-wrap: break-word;
}
.dokuwiki div.wrap_spoiler {
  margin-bottom: 1.5em;
}
.dokuwiki .wrap_clear {
  clear: both;
  line-height: 0;
  height: 0;
  font-size: 1px;
  visibility: hidden;
  overflow: hidden;
}
.dokuwiki .wrap_hide {
  display: none;
}
@media only screen and (max-width: 600px) {
  #extension__list .legend > div {
    padding-left: 0;
  }
  #extension__list .legend div.screenshot {
    margin: 0 .5em .5em 0;
  }
  #extension__list .legend h2 {
    width: auto;
    float: none;
  }
  #extension__list .legend div.linkbar {
    clear: left;
  }
  [dir=rtl] #extension__list .legend > div {
    padding-right: 0;
  }
  [dir=rtl] #extension__list .legend div.screenshot {
    margin: 0 0 .5em .5em;
  }
  [dir=rtl] #extension__list .legend div.linkbar {
    clear: right;
  }
}
div.approval {
  margin: 2em;
  margin-left: auto;
  margin-right: auto;
  width: 70% !important;
  min-height: 40px;
  clear: both;
  text-align: justify;
  vertical-align: middle;
  border-collapse: collapse;
  padding: 15px 20px 15px 20px;
  -moz-border-radius: 10px;
  -khtml-border-radius: 10px;
  border-radius: 10px;
  border: solid black 2px;
}
div.approved_yes {
  background-color: #dfd;
  background-image: url(/wiki/lib/plugins/publish/images/yes.gif);
}
div.approved_no {
  background-color: #fdd;
  background-image: url(/wiki/lib/plugins/publish/images/no.gif);
}
div.approval em {
  font-weight: bold;
}
div.approval span {
  display: block;
}
div.approved_revision {
  background-color: #dfd;
}
div.unapproved_revision {
  background-color: #fdd;
}
table.apr_table {
  width: 100%;
}
table.apr_table tr.apr_head {
  border-bottom: solid black 2px;
}
table.apr_table tr.apr_ns {
  border-bottom: solid black 1px;
}
table.apr_table tr.apr_ns td {
  text-align: center;
  font-weight: bold;
}
a.approved_diff_link {
  visibility: hidden;
}
div.approval span span.approval_date {
  display: inline;
  font-size: 80%;
}
.dokuwiki div.wrap_box,
.dokuwiki div.wrap_danger,
.dokuwiki div.wrap_warning,
.dokuwiki div.wrap_caution,
.dokuwiki div.wrap_notice,
.dokuwiki div.wrap_safety,
.dokuwiki div.wrap_info,
.dokuwiki div.wrap_important,
.dokuwiki div.wrap_alert,
.dokuwiki div.wrap_tip,
.dokuwiki div.wrap_help,
.dokuwiki div.wrap_todo,
.dokuwiki div.wrap_download {
  border: 2px solid #999;
  padding: 1em 1em .5em;
  margin-bottom: 1.5em;
}
.dokuwiki span.wrap_box,
.dokuwiki span.wrap_danger,
.dokuwiki span.wrap_warning,
.dokuwiki span.wrap_caution,
.dokuwiki span.wrap_notice,
.dokuwiki span.wrap_safety,
.dokuwiki span.wrap_info,
.dokuwiki span.wrap_important,
.dokuwiki span.wrap_alert,
.dokuwiki span.wrap_tip,
.dokuwiki span.wrap_help,
.dokuwiki span.wrap_todo,
.dokuwiki span.wrap_download {
  border: 1px solid #999;
  padding: 0 .3em;
}
.dokuwiki .wrap_hi {
  border: 1px solid #999;
}
.dokuwiki .wrap_spoiler {
  visibility: hidden;
}
.dokuwiki .wrap_pagebreak {
  page-break-after: always;
}
.dokuwiki .wrap_nopagebreak {
  page-break-inside: avoid;
}
.dokuwiki .wrap_noprint {
  display: none;
}
#search__jump {
  display: none;
}
body {
  font-family: sans-serif;
}
a {
  text-decoration: none;
}
table.inline {
  border: 1px solid #808080;
  border-collapse: collapse;
}
table.inline td,
th {
  border: 1px solid #808080;
}
pre {
  background-color: #eeeeee;
  color: #000000;
  padding: 0.3em;
}
em.u {
  font-style: normal;
  text-decoration: underline;
}
em em.u {
  font-style: italic;
}
img.medialeft {
  float: left;
}
img.mediaright {
  float: right;
}
blockquote {
  border-left: 3px solid #cccccc;
  margin: 0;
  padding: 5px 0 5px 15px;
}
.rightalign {
  text-align: right;
}
.centeralign {
  text-align: center;
}
.leftalign {
  text-align: left;
}
table.pdfheader {
  font-size: 8pt;
  border-collapse: collapse;
  border-bottom: 1px solid #000000;
  clear: both;
}
table.pdffooter {
  font-size: 8pt;
  border-collapse: collapse;
  border-top: 1px solid #000000;
  clear: both;
}
@page { size:auto; odd-header-name: html_header_odd;
even-header-name: html_header_even;
odd-footer-name: html_footer_odd;
even-footer-name: html_footer_even;
}@page :first {}@page landscape-page { size:landscape }div.dw2pdf-landscape { page:landscape-page }@page portrait-page { size:portrait }div.dw2pdf-portrait { page:portrait-page }</style></head><body><htmlpageheader name="header_odd">
<table width="100%" class="pdfheader">
    <tr>
        <td style="text-align: left">Schütte-Wiki: PlayGround</td>
        <td style="text-align: center"></td>
        <td style="text-align: right">2015/01/12 10:18</td>
    </tr>
</table>

</htmlpageheader>
<htmlpageheader name="header_even">
<table width="100%" class="pdfheader">
    <tr>
        <td style="text-align: left">Schütte-Wiki: PlayGround</td>
        <td style="text-align: center"></td>
        <td style="text-align: right">2015/01/12 10:18</td>
    </tr>
</table>

</htmlpageheader>
<htmlpagefooter name="footer_odd">
<table width="100%" class="pdffooter">
    <tr>
        <td style="text-align: left">playground:playground</td>
        <td style="text-align: center"></td>
        <td style="text-align: right">{PAGENO}/{nbpg}</td>
    </tr>
</table>

</htmlpagefooter>
<htmlpagefooter name="footer_even">
<table width="100%" class="pdffooter">
    <tr>
        <td style="text-align: left">playground:playground</td>
        <td style="text-align: center"></td>
        <td style="text-align: right">{PAGENO}/{nbpg}</td>
    </tr>
</table>

</htmlpagefooter>
<div class="dokuwiki"><a name="playgroundplayground__"></a>
<h1><bookmark content="PlayGround" level="0" /><a name="playgroundplayground__playground">PlayGround</a></h1>
<div class="level1">

<p>
<img src="/wiki/lib/images/smileys/fixme.gif" class="icon" alt="FIXME" /> ⇒ <img src="/wiki/lib/images/smileys/icon_exclaim.gif" class="icon" alt=":!:" /> ⇐
</p>

</div>
<br />
<br />
<div style="font-size: 80%; border: solid 0.5mm #DDDDDD;background-color: #EEEEEE; padding: 2mm; border-radius: 2mm 2mm; width: 100%;">
    <table width="100%">
        <tr>
            <td>
                Quelle:<br />
                <a href="http://appsrv03/wiki/">http://appsrv03/wiki/</a>&nbsp;-&nbsp;<b>Schütte-Wiki</b>
                <br /><br />
                Adresse der Seite:<br />
                <b><a href="http://appsrv03/wiki/doku.php?id=playground:playground">http://appsrv03/wiki/doku.php?id=playground:playground</a></b>
                <br /><br />
                Letztes Update der Seite: <b>2015/01/12 10:17</b>
                <br />
            </td>
            <!-- kein QR-Code:
			<td align="right">
                <img src="https://chart.googleapis.com/chart?chs=120x120&cht=qr&chl=http%3A%2F%2Fappsrv03%2Fwiki%2Fdoku.php%3Fid%3Dplayground%3Aplayground" />
            </td>  -->
        </tr>
    </table>
</div>
</div></body></html>
```
Regards
Juergen

**Comment by floweh** [comment]
Just to bump this. Still an issue. If I can help with testing please let me know where to start

**Comment by Klap-in** [comment]
@splitbrain @micgro42 do you know what kind of path are required for images to be exported fine by dw2pdf? 

Could it be related to less default settings for basedir, baseurl, canonical, userewrite, etc?

**Comment by micgro42** [comment]
During the last summer, the plugin had a major upgrade, i.e. new mpdf version. That also affected images and how they are included, see #288.
Also, the line "`FIXME => :?: ⇒ :!: ⇐ DELETEME <=`" exports perfectly fine for me. Hence, I would suggest closing this issue, unless there are reports affecting the current version of the dw2pdf plugin.

**Comment by Klap-in** [comment]
Ok, thanks for the information! Then I assume all these image issues as fixed in the last release of dw2pdf.
